src/main/scala/strawman/collection/mutable/ArrayDeque.scala

```diff
@@ -157,6 +157,12 @@ class ArrayDeque[A] private(var array: Array[AnyRef], var start: Int, var end: I
     }
   }
 
+  override def reverse = {
+    val r = new ArrayDeque[A](initialSize = size)
+    indices.foreach({i => r.appendAssumingCapacity(this(size - i - 1))})
```

Comment:
It's going to be tough for the JVM to turn this into a reverse as fast as the while loop version that simply counts up indices on the target and counts them up with an offset and check on exceeding the bounds on the current collection.

As a mockup this is fine, but can you leave a note that this can also be sped up?